dirs/dirs.go

```diff
@@ -179,7 +183,8 @@ func SetRootDir(rootdir string) {
 		DistroLibExecDir = filepath.Join(rootdir, "/usr/lib/snapd")
 	}
 
-	CoreLibExecDir = filepath.Join(rootdir, "/usr/lib/snapd")
+	CoreLibExecDir = "/usr/lib/snapd"
+	CoreSnapMountDir = "/snap"
```

Comment:
What's the line for deciding what has a prefix or not?  Why do we have BaseSnapMountDir with rootdir, but the ones above not?  Hard to guess on call sites, which makes it quite error prone.